### ISSUE TITLE
Shell Script and Bash Commands Highlight

### ISSUE THREAD

**Issue by alexr00** [atom/language-shellscript#130]
_From @agenciah1code on November 7, 2018 13:3_

After 5 years using Sublime, I am migrating to VS Code, however I have noticed a big problem in HighLight of Shell (Bash) language, variables are not colored correctly, bash commands, etc. Is there any forecast or any way to fix this?

_Copied from original issue: Microsoft/vscode#62723_

**Comment by alexr00** [comment]
_From @chrmarti on November 7, 2018 14:48_

/findDuplicates

**Comment by alexr00** [comment]
_From @vscodebot[bot] on November 7, 2018 14:48_

Potential duplicates:
- [Rendering issues when ssh'd to a Windows git bash shell. (#54682)](https://www.github.com/Microsoft/vscode/issues/54682) 
- [After update embedded bash went broken (#46855)](https://www.github.com/Microsoft/vscode/issues/46855) 
- [Git bash history does not record commands. (#39877)](https://www.github.com/Microsoft/vscode/issues/39877) 
- [There is a problem with the Korean on the terminal (git bash). (#61380)](https://www.github.com/Microsoft/vscode/issues/61380) 
- [Vs code file is not getting updated when i commit from git bash (#61428)](https://www.github.com/Microsoft/vscode/issues/61428)

**Comment by alexr00** [comment]
@agenciah1code, is this what you mean?
![image](https://user-images.githubusercontent.com/38270282/48191428-57cb7400-e345-11e8-85b0-86e721189dd5.png)
Where `file` is not colored?

It looks like this is a problem with the grammar we use from atom/language-shellscript.

**Comment by kinoute** [comment]
Any news on this? In my case, it's more about other shell functions like `ssh` that are not supported and are classified as `source.shell` as well.

Also, arguments don't have a proper color.

Sublime:
![capture d ecran 2019-01-09 a 22 37 40](https://user-images.githubusercontent.com/623763/50930142-717d5a80-145f-11e9-902f-9889d7f7bfe7.png)

VSCode:
![capture d ecran 2019-01-09 a 22 37 51](https://user-images.githubusercontent.com/623763/50930150-79d59580-145f-11e9-98d5-8fcb5e22d63d.png)

**Comment by ghubusr01** [comment]
Yes, sadly there are a lot of other shell functions/commands not classified as @kinoute said like cat, systemctl and so much more. Would be great if they would be classified somehow so they could be colorized
